btcec/schnorr/musig2/sign.go

```diff
@@ -104,65 +104,65 @@ func (p *PartialSignature) Decode(r io.Reader) error {
 
 // SignOption is a functional option argument that allows callers to modify the
 // way we generate musig2 schnorr signatures.
-type SignOption func(*signOptions)
+type SignOption func(*SignOptions)
 
-// signOptions houses the set of functional options that can be used to modify
+// SignOptions houses the set of functional options that can be used to modify
 // the method used to generate the musig2 partial signature.
-type signOptions struct {
-	// fastSign determines if we'll skip the check at the end of the
+type SignOptions struct {
```

Comment:
Why does any of this need to be exported here? Having these fields be private enables a type of encapsulation and information hiding here. 

API design here is that new options are added to the package, as they're a bit more "curated", and exported via the functions like `WithFastSign`.

Reply:
check out the link in the PR description. It is used in the `RPCKeyRing.SignMuSig2` method [here](https://github.com/lightningnetwork/lnd/pull/8163/commits/b21177a36812b4d77d08fc1e9eeb33a30b63aadb). 

We cant pass the options as is over gRPC, so if we want to be able to do MuSig2 signing over the signer client, then we need to be able to inspect the signing options

Reply:
Ah, in the past, what I ended up doing there was to just sorta replicate the option types in an unrolled manner, then you map then to `SignOption` at the latest stage. We had to do this for the `musig2` options as well.